dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java

```diff
@@ -287,23 +223,76 @@ public void createPullReq(GHRepository origRepo,
+     *
+     * @param repository the repository which should come from an expected parent
+     * @param gitForkBranch the branch name which was derived from the image to update
+     * @return Optional GHPullRequest if we've found one
+     */
+    public Optional<GHPullRequest> getPullRequestForImageBranch(GHRepository repository, GitForkBranch gitForkBranch) {
+        PagedIterable<GHPullRequest> pullRequests =
+                repository.queryPullRequests().state(GHIssueState.OPEN).head(gitForkBranch.getBranchName()).list();
+        for (GHPullRequest pullRequest : pullRequests) {
+            // There can be only one since it is based on branch.
+            return Optional.of(pullRequest);
         }
+        return Optional.empty();
+    }
 
-        for (GHPullRequest pullRequest : pullRequests) {
-            GHUser user = pullRequest.getHead().getUser();
-            if (myself.equals(user) && pullRequest.getBody().equals(Constants.PULL_REQ_ID)) {
-                return pullRequest;
+    public void createOrUpdateForkBranchToParentDefault(GHRepository parent, GHRepository fork, GitForkBranch gitForkBranch) throws IOException {
+        GHBranch parentBranch = parent.getBranch(parent.getDefaultBranch());
+        String sha1 = parentBranch.getSHA1();
+        GHBranch forkBranch = fork.getBranches().get(gitForkBranch.getBranchName());
+        String branchRefName = String.format("refs/heads/%s", gitForkBranch.getBranchName());
+        if (forkBranch == null) {
+            fork.createRef(branchRefName, sha1);
+        } else {
+            fork.getRef(branchRefName).updateTo(sha1, true);
+        }
+    }
+
+    /**
+     * Determines whether the user we're logged in as is the repo owner
+     *
+     * @param repo the repo to check
+     * @return are we the repo owner?
+     * @throws IOException when attempting to check our user details
+     */
+    public boolean thisUserIsOwner(GHRepository repo) throws IOException {
+        String repoOwner = repo.getOwnerName();
+        GHMyself myself = gitHubUtil.getMyself();
+        if (myself == null) {
+            throw new IOException("Could not retrieve authenticated user.");
+        }
+        String myselfLogin = myself.getLogin();
+        return repoOwner.equals(myselfLogin);
+    }
+
+    /**
+     * Get GHContents for a provided image in the provided GitHub org.
+     *
+     * @param org GitHub organization
+     * @param img image to find
+     */
+    public Optional<PagedSearchIterable<GHContent>> getGHContents(String org, String img)
+            throws IOException, InterruptedException {
+        PagedSearchIterable<GHContent> contentsWithImage = null;
+        for (int i = 0; i < 5; i++) {
+            contentsWithImage = findFilesWithImage(img, org);
+            if (contentsWithImage.getTotalCount() > 0) {
+                break;
+            } else {
+                Thread.sleep(1000);
```

Comment:
`TimeUnit.SECONDS.sleep(1);`

Reply:
Change still not made?

Reply:
Bah... There were a bunch... Corrected that crust